pkg/image/api/types.go

```diff
@@ -60,6 +84,69 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureState holds a result of signature's server-side check.
+type SignatureState string
+
+// SignatureForImage describes relation of the signature to the image containing it.
+type SignatureForImage string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side signature check. IOW a time of the last update of SignatureState.
+	Checked *unversioned.Time
+	// Says whether the signature matches image object containing it.
+	ForImage SignatureForImage
```

Comment:
I don't find this discussion that important since it doesn't affect the code. But still I'd like to clarify. `ForImage = "match"` is set based on `image.Signature[i].ImageHash == image.Name` as long as the signature can be parsed and has all the mandatory fields filled. Therefore I believe the `ForImage` should `"match"` for `State` in any of `{trusted, expired, revoked}`.

Of course a policy for running/examining the signed content based solely on `ForImage` doesn't make sense.